rpcs3/Emu/Cell/lv2/sys_net.cpp

```diff
@@ -191,7 +235,10 @@ struct network_thread
 		{
 			// Wait with 1ms timeout
 #ifdef _WIN32
-			::WSAPoll(fds, socklist.size(), 1);
+			{
+				reader_lock lock(id_manager::g_mutex);
```

Comment:
This is bad idea to lock it here. Consider adding 4th argument to the function with a reference to the socket array (like socklist few lines earlier).

Reply:
I moved all the is_connecting logic to the same place where it setups the fd array in each function so it should be optimal performance wise now(no extra lookup) even if it leads to a bit more #ifdef _WIN32 in the code.

Reply:
you misunderstood, Its possible to pass an array of 1024 raw/shared pointers here. (ptr type is based on template deduction, passed as a pointer to pointers)

Reply:
the idea is to not use idm inside the function.

Reply:
I understand I could use socklist just in the thread loop but the change I made also removes any idm lookup/calls for all the fds at the beginning for all instances of WSAPoll calls. The lambda setting is_connecting to false should be very rarely called anyway and I don't think it's an issue.

I can make a special function for the loop if it's really deemed necessary but I don't think it is.